sequencer/src/bin/cdn-broker.rs

```diff
@@ -0,0 +1,123 @@
+//! The following is the main `Broker` binary, which just instantiates and runs
+//! a `Broker` object.
+
+use anyhow::{Context, Result};
+use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
+use cdn_broker::{Broker, Config, ConfigBuilder};
+use clap::Parser;
+
+use hotshot::traits::implementations::{KeyPair, ProductionDef, WrappedSignatureKey};
+use hotshot_types::traits::node_implementation::NodeType;
+use hotshot_types::traits::signature_key::SignatureKey;
+use sequencer::SeqTypes;
+use sha2::Digest;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+/// The main component of the push CDN.
+struct Args {
+    /// The discovery client endpoint (including scheme) to connect to.
+    /// With the local discovery feature, this is a file path.
+    /// With the remote (redis) discovery feature, this is a redis URL (e.g. `redis://127.0.0.1:6789`).
+    #[arg(short, long, env = "ESPRESSO_CDN_BROKER_DISCOVERY_ENDPOINT")]
+    discovery_endpoint: String,
+
+    /// Whether or not metric collection and serving is enabled
+    #[arg(
+        long,
+        default_value_t = false,
+        env = "ESPRESSO_CDN_BROKER_METRICS_ENABLED"
+    )]
+    metrics_enabled: bool,
+
+    /// The IP to bind to for externalizing metrics
+    #[arg(
+        long,
+        default_value = "127.0.0.1",
+        env = "ESPRESSO_CDN_BROKER_METRICS_IP"
+    )]
+    metrics_ip: String,
+
+    /// The port to bind to for externalizing metrics
+    #[arg(long, default_value_t = 9090, env = "ESPRESSO_CDN_BROKER_METRICS_PORT")]
+    metrics_port: u16,
+
+    /// The user-facing address to bind to for connections from users
+    #[arg(
+        long,
+        default_value = "0.0.0.0:1738",
+        env = "ESPRESSO_CDN_BROKER_PUBLIC_BIND_ADDRESS"
+    )]
+    public_bind_address: String,
+
+    /// The user-facing address to advertise
+    #[arg(
+        long,
+        default_value = "local_ip:1738",
```

Comment:
Took a brief look at the push cdn code. So `local_ip` is a magic value that will be resolved to a local IP? What will it choose if there are multiple network interfaces? Would we ever not want this to be set to `local_ip`?

Reply:
Yessir, downstream it is resolved to whatever `local_ip::local_ip_address()` resolves to. I made it like this so it's easier to specify it if it's unknown and not set (e.g. in AWS).

If there are multiple networking interfaces, my thought is to use `0.0.0.0` to bind to all of them.  For example, we would want it (`public_advertise_address` specifically) to be `0.0.0.0` on a fully public deployment that does not use load balancers